### PR TITLE
Refactoring `external.py`

### PR DESCRIPTION
**Pull request by abidlabs**
While working with `external.py`, I noticed that the module had gotten quite bloated and hard-to-read, so I refactored the file. Major changes:

* Moved many helper functions from `external.py` to `external_utils.py`
* Added some typing and changed `test_external.py` to use pytest-style tests
* Moved the functions to create an `Interface` from a `transformers.Pipeline` object to a separate `pipelines.py` and its test to a `test_pipelines.py`

As part of this refactoring, I noticed that we never tested `gr.Interface.from_pipeline()` method (the original test was a dud!). So I've added a real test for the function -- no other code logic was changed.

### CODE REVIEW

File: gradio/external.py [hunk 1]
```diff
@@ -393,7 +310,7 @@ def query_huggingface_api(*params):
     return interface
 
 
-def get_spaces(space_name: str, api_key: str | None, alias: str, **kwargs) -> Blocks:
+def from_spaces(space_name: str, api_key: str | None, alias: str, **kwargs) -> Blocks:
```
Comment (freddyaboulton) [inline gradio/external.py:313]: Unrelated to this change but if `iframe_url` is none we should raise an exception

File: gradio/dataclasses.py [hunk 2]
```diff
@@ -1,3 +1,5 @@
+from __future__ import annotations
```
Comment (freddyaboulton) [inline gradio/dataclasses.py:1]: How come you added this?

File: gradio/external.py (hunk 1 above)
Comment (abidlabs) [reply gradio/external.py:313]: Ah yes we can do that

**Comment by abidlabs** [comment]
Thanks for the review @freddyaboulton! Will address the points and then merge

**Pull request merged**